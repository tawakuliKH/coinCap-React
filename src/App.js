import {
  BrowserRouter, Route, Routes,
} from 'react-router-dom';
import Coins from './components/Coins';
import CoinsDetails from './components/CoinsDetails';
import Header from './components/Navbar';

function App() {
  return (
    <div className="App">
      <BrowserRouter>

        <Header />

        <Routes>
          <Route path="/" element={<Coins />} />
          <Route path="/details/:id" element={<CoinsDetails />} />
        </Routes>

      </BrowserRouter>
    </div>
  );
}

export default App;
