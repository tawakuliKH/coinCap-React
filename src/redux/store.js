import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import coinsSlice from './coins/CoinsSlice';
import coinsDetailsSlice from './coinsDetails/coinsDetailsSlice';

const store = configureStore({
  reducer: {
    coins: coinsSlice,
    coinsDetails: coinsDetailsSlice,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
