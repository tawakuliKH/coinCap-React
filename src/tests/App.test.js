import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import App from '../App';
import store from '../redux/store';
import '@testing-library/jest-dom';


test('renders Coins component with Redux Provider', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  const headerComponent = getByTestId('app-component');
  expect(headerComponent).toBeInTheDocument();
});
