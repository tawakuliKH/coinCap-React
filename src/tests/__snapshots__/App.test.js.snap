// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Coins component with Redux Provider 1`] = `
<div
  className="App"
  data-testid="app-component"
>
  <header
    className="header"
    data-testid="navbar-component"
  >
    <h3>
      Crypto Coins
    </h3>
  </header>
  <section
    className="coin-section"
    data-testid="coins-component"
  >
    <div
      className="coin-topic"
    >
      <h2>
        Cryptocurrency Market
      </h2>
      <p>
        (
        0
         
        Famouse Crypto Coins)
      </p>
      <div
        className="search-bar-container"
      >
        <input
          className="search-bar"
          onChange={[Function]}
          placeholder="Search"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="card-container"
    />
  </section>
</div>
`;
